### ISSUE TITLE
Build fail on verifying python requirement

### ISSUE THREAD

**Issue by danilosampaio** [atom/atom#16907]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

**Some steps are not applicable to the build scenario.** :warning:

### Description

The bootstrap script doesn't verify properly the machine requirements,
throwing a unhandled error when **python** is not installed.

```C:\Users\jdansilv\github\atom\script>bootstrap
Node:   v8.10.0
Npm:    v5.6.0
child_process.js:624
    throw err;
    ^

Error: spawnSync python ENOENT
    at _errnoException (util.js:1022:11)
    at spawnSync (child_process.js:579:20)
    at Object.execFileSync (child_process.js:616:13)
    at verifyPython (C:\Users\jdansilv\github\atom\script\lib\verify-machine-requirements.js:55:29)
    at module.exports (C:\Users\jdansilv\github\atom\script\lib\verify-machine-requirements.js:13:5)
    at Object.<anonymous> (C:\Users\jdansilv\github\atom\script\bootstrap:20:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
```

**Suggestion:** verify the existence of 'python' in the `PATH` variable, before to set `pythonExecutable = 'python'`, in the file `script/lib/verify-machine-requirements.js`, line 51:

Ex: 
```js
if (process.env.PATH.match(/python(;|$)/)) {
	pythonExecutable = 'python'
} else {
	throw new Error(
	  `Python 2.7 is required to build Atom.`
        )
}
```

### Steps to Reproduce

1. git clone git@github.com:<em>your-username</em>/atom.git
2. cd <em>where-you-cloned-atom</em>
3. script/bootstrap

**Expected behavior:** 

print a handled error: Python 2.7 is required to build Atom.

**Actual behavior:** 

print a unhandled error: Error: spawnSync python ENOENT

**Reproduces how often:** 100%

### Versions

Windows 10
Node:   v8.10.0
Npm:    v5.6.0
Atom: 1.26.0-dev

**Comment by danilosampaio** [comment]
A minor adjust on path checking:

```js
if (process.env.PATH.match(/python[0-9]*(;|$)/)) {
  ...
```

**Comment by Arcanemagus** [comment]
It looks like this should potentially be solved in https://github.com/atom/atom/pull/16885 if I'm not mistaken @smashwilson?

**Comment by smashwilson** [comment]
@Arcanemagus Hopefully, although I haven't touched `verify-machine-requirements.js` there yet.

@danilosampaio that'll get you past the requirement verification, but does it let you complete the build? I hit problems with node-gyp during apm installation when the Python 2 binary wasn't called exactly `python`.

**Comment by danilosampaio** [comment]
@Arcanemagus / @smashwilson Since the `script/bootstrap` runs earlier than `script/build`, it could just check python installation ([85fb56](https://github.com/danilosampaio/atom/commit/85fb560770f6f52d6a050633b5b0bdffa3e926d7)), and conflicts related to the python versions could be fixed in #16885. It sounds like different problems to me.
